### PR TITLE
Support for auto hiding controls after 5 seconds

### PR DESCRIPTION
**Pull request by codeimpl**
Make the remote controls disappear automatically after 5 seconds of the
user touching the screen or updating the screen.

Refs: #610

### CODE REVIEW

File: src/org/triple/banana/remote_control/RemoteControlLayout.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.widget.RelativeLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class RemoteControlLayout extends RelativeLayout {
+    Callback mCallback;
```
[the remark this answers is not in the document]
Comment (codeimpl) [reply src/org/triple/banana/remote_control/RemoteControlLayout.java:17]: done.

File: src/org/triple/banana/remote_control/RemoteControlLayout.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    public interface Callback {
+        void onInterceptTouchEvent();
+    }
+
+    public RemoteControlLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public void setCallback(Callback callback) {
```
[the remark this answers is not in the document]
Comment (codeimpl) [reply src/org/triple/banana/remote_control/RemoteControlLayout.java:27]: done

File: src/org/triple/banana/remote_control/RemoteControlService.java [hunk 3]
```diff
@@ -26,8 +27,14 @@
...
     private boolean mWasPipMode;
     private boolean mWasPlaying;
+    private final int TIME_OUT = 5000;
 
     private MediaController mMediaController = MediaController.instance;
+    private Runnable mControlHideRunnable = () -> {
+        mViewModel.getEditor().setControlsVisibility(false);
+        mViewModel.commit();
+    };
+    private Handler mHandler = new Handler();
```
[the remark this answers is not in the document]
Comment (codeimpl) [reply src/org/triple/banana/remote_control/RemoteControlService.java:37]: done

File: src/org/triple/banana/remote_control/RemoteControlService.java [hunk 4]
```diff
@@ -26,8 +27,14 @@
     private RemoteControlViewImpl mView = new RemoteControlViewImpl(this);
     private boolean mWasPipMode;
     private boolean mWasPlaying;
+    private final int TIME_OUT = 5000;
 
     private MediaController mMediaController = MediaController.instance;
+    private Runnable mControlHideRunnable = () -> {
```
[the remark this answers is not in the document]
Comment (codeimpl) [reply src/org/triple/banana/remote_control/RemoteControlService.java:33]: done

File: src/org/triple/banana/remote_control/RemoteControlService.java [hunk 5]
```diff
@@ -26,8 +27,14 @@
     private RemoteControlViewImpl mView = new RemoteControlViewImpl(this);
     private boolean mWasPipMode;
     private boolean mWasPlaying;
+    private final int TIME_OUT = 5000;
```
[the remark this answers is not in the document]
Comment (codeimpl) [reply src/org/triple/banana/remote_control/RemoteControlService.java:30]: done

**Review by romandev — APPROVED** [review]
LGTM

**Pull request merged**